matminer/featurizers/base.py

```diff
@@ -151,6 +155,84 @@ def set_chunksize(self, chunksize):
     def chunksize(self):
         return self._chunksize if hasattr(self, '_chunksize') else None
 
+    def precheck_dataframe(self, df, col, return_frac=True, inplace=False) \
+            -> [float, pd.DataFrame]:
+        """
+        Precheck an entire dataframe. Subclasses wanting to use precheck
+        functinoality should not override this method, they should override
+        precheck (unless the entire df determines whether single entries pass
+        or fail a precheck).
+
+        Prechecking should be a quick and useful way to check that for a
+        particular dataframe (set of featurizer inputs), the featurizer is:
+
+            1. in scope, and/or...
+            2. robust to errors and/or...
+            3. any other reason you would not practically want to use this
+                featurizer in on this dataframe.
+
+        By prechecking before featurizing, you can avoid applying featurizers
+        to data that will ultimately fail, return unreliable numbers, or
+        are out of scope. Prechecking is also a good time to throw/observe
+        warnings (such as long runtime warnings!).
+
+        Args:
+            df (pd.DataFrame): A dataframe
+            col (str): The column in df specifying the input to the featurizer.
```

Comment:
Can we allow this to be a list of strings, in the case of featurizers that take >1 argument as input?

Reply:
Sure